### DISCUSSION TITLE
Creating HARs.

### DISCUSSION THREAD

**Discussion by beardie79** [Steinerd/Loupedeck.WebhooksPlugin#11] [General]
_"There are numerous ways to create them, or export them in many different applications. Including Postman, Telerik Fiddler2, and even most browsers DevTools will allow you to export/copy web requests as HARs."_

May I ask if you have any resources / sources on how to learn how to generate these HARs? I am migrating from a StreamDeck where I could just post a nabu.casa URL from HomeAssistant.

**Comment by steinerd** [comment]
I kinda just found the standard and rolled with it.

The learning curve for how this plugin is using them is INSANELY small. 

This plugin is only looking at the entries respective 

1. `comment` for the name it displays in Loupedeck
2. `request.postData.text` for the payload
3. `request.postData.mimeType` for the Content-Type (normally `application/json` or `text/xml`)
4. `url` for the actual endpoint it invokes (put `?query=string` values in this field)
5. `method` which tells the plugin how to deliver the HTTP request to the endpoint. (GET/POST/PUT/DELETE...)

You can either create multiple HAR files in the directory explained in the documentation, or stick all of the endpoints in the same HAR file, since `entries` in an array, however that can become difficult to read if it gets too many entries. 

Below is an example of a hollowed-out HAR file, just replace the items with curly braces with the desired values {{...}} (be sure to remove the curly-braces as well). 

```json
{
    "log": {
        "pages": [],
        "comment": "",
        "entries": [
            {
                "comment": "{{PUT LINK NAME HERE}}",
                "time": 0,
                "serverIPAddress": "127.0.0.1",
                "connection": "ClientPort:0;EgressPort:62999",
                "request": {
                    "headersSize": 0,
                    "postData": {
                        "text": "{{PUT ESCAPED POST DATA PAYLOAD HERE}}",
                        "mimeType": "{{PUT CONTENT-TYPE HERE}}"
                    },
                    "queryString": [],
                    "headers": [
                        {
                            "name": "Content-Type",
                            "value": "application/json"
                        },
                        {
                            "name": "Accept",
                            "value": "application/json"
                        }                        
                    ],
                    "bodySize": 0,
                    "url": "{{PUT URL HERE}}",
                    "cookies": [],
                    "method": "{{PUT GET/POST/PUT/DELETE HERE}}",
                    "httpVersion": "HTTP/1.1"
                },
                "timings": {
                    "blocked": -1,
                    "ssl": 0,
                    "receive": 0,
                    "wait": 0,
                    "dns": 0,
                    "send": 0,
                    "connect": 0
                },
                "response": { },
                "startedDateTime": "2022-04-08T16:33:37.2014031-04:00",
                "cache": {}
            }
        ],
        "creator": {
            "name": "Fiddler",
            "comment": "https://fiddler2.com",
            "version": "5.0.20211.51073"
        },
        "version": "1.2"
    }
}
```
